homeassistant/components/rointe/coordinator.py

```diff
@@ -0,0 +1,105 @@
+"""Provides the Rointe DataUpdateCoordinator."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from rointesdk.device import RointeDevice
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, LOGGER, PLATFORMS
+from .device_manager import RointeDeviceManager
+
+ROINTE_API_REFRESH_INTERVAL = timedelta(seconds=15)
+
+
+class RointeDataUpdateCoordinator(DataUpdateCoordinator[dict[str, RointeDevice]]):
+    """Rointe data coordinator."""
+
+    def __init__(
+        self, hass: HomeAssistant, device_manager: RointeDeviceManager
+    ) -> None:
+        """Initialize Rointe data updater."""
+        self.device_manager = device_manager
+        self.unregistered_keys: dict[str, dict[str, RointeDevice]] = {}
+
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=ROINTE_API_REFRESH_INTERVAL,
+        )
+
+        self.unregistered_keys = {platform: {} for platform in PLATFORMS}
```

Comment:
What's a key?

Reply:
In that context a "key" is the unique identifier of each device. It's determined by the manufacturer's API

Reply:
By the way, do you have discord? Can you maybe pop me a message? (@joostlek)

Reply:
```suggestion
```